substrate/frame/contracts/src/lib.rs

```diff
@@ -656,7 +656,64 @@ pub mod pallet {
 				"Debug buffer should have minimum size of {} (current setting is {})",
 				MIN_DEBUG_BUF_SIZE,
 				T::MaxDebugBufferLen::get(),
-			)
+			);
+
+			// Validators have configured runtime memory for 512MB. They need to keep in it
+			// max_runtime_mem, PoV in multiple copies (1x encoded + 2x decoded), and storage
+			// including events. The assumption is that max_runtime_mem + storage/events can be
```

Comment:
It is my assumption, please comments if it make sense. PoV in total is less than 50MB (including copies), I think.
There are still storage transaction layer, but after last improvements most probably they will not keep storage copies.